packages/flutter_tools/test/general.shard/android/adb_log_reader_test.dart

```diff
@@ -79,6 +79,30 @@ void main() {
     expect(processManager.hasRemainingExpectations, false);
   });
 
+  testWithoutContext('AdbLogReader calls adb logcat with expected flags when requesting past logs', () async {
```

Comment:
On a scale of 1 to 10, how difficult was this test to write compared to other tool tests or framwork tests? Asking for a friend

Reply:
what are you asking specifically? the fakeprocessmanager expectations thing?

Reply:
mostly yeah

Reply:
oh, I think I would have been confused. Consider the mockito or xctest API, the word verify or expect clearly appears in the API name so it's clear you're stacking up expectations before running. 

It's also a mock not a fake no? https://blog.pragmatists.com/test-doubles-fakes-mocks-and-stubs-1a7491dfa3da

But in this case, I just copied the test above so difficulty was 1 :)

Reply:
The are Dart specific considerations - i.e. MockFoo needs to extend a mockito Mock base class, otherwise you'll can be very confused if `when()` and `verify()` stop working. 

I'm not sure that this isn't a fake though. it might be a Spy? From the perspective of the tool code all the process manager does is create process objects

Reply:
"mock" isn't a branded word though. Mockito happening to have its convenience class be called Mock shouldn't mean they reserve the technical word "mock" across the board. See https://github.com/flutter/flutter/blob/master/packages/flutter/test/material/icon_button_test.dart#L15, https://github.com/flutter/flutter/blob/master/packages/flutter/test/material/text_field_test.dart#L23, https://github.com/flutter/flutter/blob/master/packages/flutter/test/painting/image_stream_test.dart#L49.

Dart mockito also originates from Java mockito where they didn't reserve the word "mock" generally.

Reply:
It is effectively a branded word in Dart. Sorry!

Reply:
What does that mean exactly? It's already effectively not true in our codebase.

Reply:
See also https://github.com/flutter/flutter/issues/33274

Reply:
Replied in the referenced bug